gulpfile.babel.js

```diff
@@ -0,0 +1,264 @@
+
+const isProdBuild = () => process.argv.filter(val => val.toLowerCase().indexOf('-prod') !== -1).length > 0;
+const pkgJson = require('./package.json');
+const server = browserSync.create();
+const mainDirectories = {
+	dev: './server/',
+	dist: './dist/'
+};
+
+const settings = {
+	sources: {
+		markup: './src/*.html',
+		styles: './src/assets/less/**/*.less',
+		stylesEntryPoint: './src/assets/less/index.less',
+		scripts: './src/app/**/*.js',
+		images: './src/assets/img/**/*.{png,jpg,gif,svg}',
+		externalCss: pkgJson.bootstrapKickstart.bundleCSS,
+		externalJs: pkgJson.bootstrapKickstart.bundleExternalJS,
+		staticFiles: pkgJson.bootstrapKickstart.includeStaticFiles
+	},
+	destinations: {
+		dev: {
+			markup: `${mainDirectories.dev}`,
+			styles: `${mainDirectories.dev}assets/css/`,
+			scripts: `${mainDirectories.dev}app/`,
+			images: `${mainDirectories.dev}assets/img/`,
+			libs: `${mainDirectories.dev}libs/`
+		},
+		prod: {
+			markup: `${mainDirectories.dist}`,
+			styles: `${mainDirectories.dist}assets/css/`,
+			scripts: `${mainDirectories.dist}app/`,
+			images: `${mainDirectories.dist}assets/img/`,
+			libs: `${mainDirectories.dist}libs/`
+		}
+	},
+	autoPrefix: [
+		'> 1%',
+		'last 3 version',
+		'ie 8',
+		'ie 9',
+		'Firefox ESR',
+		'Opera 12.1'
+	]
+};
+
+function onError(err) {
+	notify({
+		title: 'Gulp Task Error',
+		subtitle: 'Plugin: <%= error.plugin %>',
+		message: 'Check the console.'
+	}).write(err);
+
+	console.log(err.toString());
+
+	this.emit('end');
+}
+
+export function clean() {
+	if (isProdBuild()) {
+		return del(['dist']);
+	}
+	return del(['server']);
+}
+
+export function styles() {
+	if (isProdBuild()) {
+		return gulp.src(settings.sources.stylesEntryPoint)
+			.pipe(plumber({errorHandler: onError}))
+			.pipe(less({
+				plugins: [new Autoprefix({
+					browsers: settings.autoPrefix
+				})]
+			}))
+			.pipe(cleanCss())
+			.pipe(rename({
+				baseName: 'index',
+				suffix: '.min'
+			}))
+			.pipe(gulp.dest(settings.destinations.prod.styles))
+			.pipe(rename('index.uncss.min.css'))
+			.pipe(uncss({
+				html: [settings.sources.markup]
+			}))
+			.pipe(gulp.dest(settings.destinations.prod.styles));
+	}
+	return gulp.src(settings.sources.stylesEntryPoint)
+		.pipe(plumber({errorHandler: onError}))
+		.pipe(sourcemaps.init())
+		.pipe(less({
+			plugins: [new Autoprefix({
+				browsers: settings.autoPrefix
+			})]
+		}))
+		.pipe(sourcemaps.write('./'))
+		.pipe(gulp.dest(settings.destinations.dev.styles));
+}
+
+export function images() {
+	if (isProdBuild()) {
+		return gulp.src(settings.sources.images)
+			.pipe(imagemin())
+			.pipe(gulp.dest(settings.destinations.prod.images));
+	}
+	return gulp.src(settings.sources.images)
+		.pipe(gulp.dest(settings.destinations.dev.images));
+}
+
+export function clientScripts() {
+	if (isProdBuild()) {
+		return gulp.src(settings.sources.scripts)
+			.pipe(rollup({
+				entry: './src/app/index.js'
+			}))
+			.pipe(babel())
+			.pipe(uglify())
```

Comment:
UglifyJS will fail with all es2015+ things like destructuring, class, default parameters...
Since we instructed babel to transpile as "babel-preset-env" it might at some point drop some of these things in the future and in turn will make uglify fail. Uglify2 has a harmony branch which is not released yet as a stable build but can be used to mitigate this problem. I used that branch in production for over half a year now without any limitations.